src/osp/UserInputHandler.cpp

```diff
@@ -252,8 +252,14 @@ void UserInputHandler::config_register_control(std::string const& name,
     bool holdable,
     std::vector<ButtonVarConfig> vars)
 {
-    m_controlConfigs[name] = ButtonConfig{ std::move(vars), holdable,
-                                          false, 0 };
+    m_controlConfigs.insert_or_assign(name, ButtonConfig{ std::move(vars), holdable, false, 0 });
+}
+
+void UserInputHandler::config_register_control(std::string const&& name,
```

Comment:
There is no such thing as const&&.

Reply:
Basically I was wanting you to make two functions.
```
void UserInputHandler::config_register_control(std::string const& name, bool holdable, std::vector<ButtonVarConfig> vars)
{
    m_controlConfigs.insert_or_assign(name, ButtonConfig{ std::move(vars), holdable, false, 0 });
}
void UserInputHandler::config_register_control(std::string && name, bool holdable, std::vector<ButtonVarConfig> vars)
{
    m_controlConfigs.insert_or_assign(std::move(name), ButtonConfig{ std::move(vars), holdable, false, 0 });
}
```

The reason for this is that insert_or_assign does not move from the first parameter if the item in question already exists, so we don't want to take the parameter by value.
Instead we want to take the parameter by const-& OR &&, so that the caller of the function gets to decide whether to use move or not.